File: swift/example/Sources/VanillaDiExample/FooBarUser.swift
class FooBarUser {
    typealias Dependency = (
        foo: Foo,
        bar: Bar
    )


    private let dependency: Dependency


    init(dependency: Dependency) {
        self.dependency = dependency
    }


    func danceWithDependencies() {
        let (foo, bar) = self.dependency

        bar.doSomething(value: foo.getSomething())
    }
}

File: ts/example/lib/foo_bar_user.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class FooBarUser {
    constructor(dependency) {
        this.dependency = dependency;
    }
    danceWithDependencies() {
        const { foo, bar } = this.dependency;
        const fooValue = foo.getSomething();
        bar.doSomething(fooValue);
    }
}
exports.FooBarUser = FooBarUser;
